### PR TITLE
Add optional pin code support (Legacy pairing mode)

### PR DESCRIPTION
**Pull request by tmigone**
Added optional pin code support (Legacy pairing mode)
Minor changes to README.md

Tested on Raspberry Pi 3 and 4.

Connects-to: #20
Change-type: major
Signed-off-by: Tomás Migone <tomas@balena.io>

### CODE REVIEW

**Comment by barhun** [comment]
This quite resembles what I originally implemented in #35 except that your code changes the pairing mode in `start.sh`. I'll report the results after I test this on my rPi2 using a Bluetooth dongle.

**Comment by barhun** [comment]
> This quite resembles what I originally implemented in #35 except that your code changes the pairing mode based on a condition check in `start.sh`. I'll report the results after I test this on my rPi2 using a Bluetooth dongle.

**Comment by AlexProgrammerDE** [comment]
https://github.blog/2016-03-30-protected-branches-improvements/ has a solution for: This branch is out-of-date with the base branch. Hope it helps.

**Comment by barhun** [comment]
This works on my Raspbery Pi 2 with a Bluetooth dongle; yet when no PIN code is set, it still asks me for a PIN code to complete the pairing process. This is the same behavior as it was before this implementation. The following output is printed to logs:

`Can't set Simple Pairing mode on hci0: Input/output error (5)`

As we're priotirizing rPi 0/3/4 as they've got built-in Bluetooth, this seems to be not a big issue. I think this should be merged as is. (:

**Comment by AlexProgrammerDE** [comment]
https://askubuntu.com/questions/774652/bluetooth-keyboard-problem

**Comment by tmigone** [comment]
Thanks for taking the time @barhun. When no PIN is set it should default to the exact same behavior as before, so what you report makes sense. As you say, we want to prioritize the newer devices with built-in Bluetooth but that does not mean we can't add support for the RPi 2 as well. I don't have currently a working bluetooth dongle, but when I do i'll revisit this to see if I can fix it.

Again, thanks to both of you for your contributions. 
Let's get @chrisys to approve this and we'll merge it.

File: README.md [hunk 1]
```diff
@@ -50,13 +56,12 @@ Secondly, balenaSound will play connection/disconnection notification sounds at
 
 **Note:** these variables should be defined as integer values without the `%` symbol.
 
-### Deploy this application
+### Set bluetooth pin code
+
+By default, balenaSound bluetooth will connect using Secure Simple Pairing mode with no pin code required. If you would like to override this, you can define the `BLUETOOTH_PIN_CODE` environment variable with the pin code you desire.
```
Comment (chrisys) [inline README.md]: Can we indicate to the user what format of pin code is required? I.e. 4 digit numeric (or whatever it happens to be)

Comment (AlexProgrammerDE) [reply README.md]: @tmigone you can copy and Paste this: `**Note:** You can use up to 16 character as the pin code. They can be alphanumeric (A-Z, a-z, 1-9, !%§, etc.).` it is from my closed PR.

Comment (AlexProgrammerDE) [reply README.md]: And if you want can you use this picture:
pin-code-config

Comment (tmigone) [reply README.md]: @chrisys Updated readme to specify pin code format and also added code to check if pin is valid. 
@AlexProgrammerDE you are correct that pin code can be almost anything up to 16 character's, but i've seen some devices that only allow numeric input, so I limited the PIN code to what the bluetooth spec says.

**Comment by AlexProgrammerDE** [comment]
@tmigone i think you need to put the new commits of me into your fork/branch . Then can @chrisys perform a merge.

**Comment by tmigone** [comment]
Yes thank you, Chris is on vacation at the moment so it will take a couple more days to get this approved, but we will get there!

**Comment by AlexProgrammerDE** [comment]
How about xginn8 and shaunmulligan?

**Comment by tmigone** [comment]
Hey @chrisys, i did experience what you are describing on macOS but it only lasted for one or two seconds. What version on macOS are you in? 
I'll merge it now and continue to monitor/test on more macOS devices, but it looks like apple does not like legacy bluetooth haha.

File: README.md (hunk 1 above)
Comment (florisre) [reply README.md]: does this actually work? I have set this as an environnment variable in Balena Dashboard but even after a reboot and a restart I can still connect to the device without entering the PIN.

**Comment by florisre** [comment]
Hi. I am still having issues with this, on both iOS and elementary OS. On my iPhone, the PIN dialogue shows up but once I enter the PIN it times out. My computer running elementary simply fails to connect, without any PIN request.

**Pull request merged**